sound/soc/sof/sof-audio.c

```diff
@@ -669,28 +669,17 @@ int sof_set_up_pipelines(struct device *dev, bool verify)
 }
 
 /*
- * This function doesn't free widgets during suspend. It only resets the set up status for all
- * routes and use_count for all widgets.
+ * Helper to tear down the pipelines. This should be called to free widgets
+ * when needed, e.g. at suspend, or critical error hit during initialization.
  */
-int sof_tear_down_pipelines(struct device *dev, bool verify)
+int sof_tear_down_pipelines(struct device *dev)
 {
 	struct snd_sof_dev *sdev = dev_get_drvdata(dev);
 	struct snd_sof_widget *swidget;
 	struct snd_sof_route *sroute;
 	int ret;
 
-	/*
```

Comment:
I have mixed feelings about this change. It makes sense logically but I also think that there is no need to send all the IPC during suspend when we know that we're going to power off the cores anyway. Also, when experimenting with this in the past, I noticed that the IPC to free the widgets during suspend in variably times out